homeassistant/components/light/tplink.py

```diff
@@ -64,24 +68,26 @@ class TPLinkSmartBulb(Light):
     def __init__(self, smartbulb: 'SmartBulb', name):
         """Initialize the bulb."""
         self.smartbulb = smartbulb
-
-        # Use the name set on the device if not set
-        if name is None:
-            self._name = self.smartbulb.alias
-        else:
+        self._name = None
+        if name is not None:
             self._name = name
```

Comment:
Was about to comment that this should be moved into `update()` before noticing you did exactly that, nice one!